File: src/TCG/SentenceTranslator/GotoTranslator.cpp
#include "Common/Common.h"
#include "TCG/SentenceTranslator/GotoTranslator.h"
#include "TCG/ASM.h"
#include "TCG/SymbolManager.h"
#include "TCG/ConstructASM.h"


ASMLines GotoTranslator::SentenceTranslate_(SymbolManager& SymbolManager_, TACLine& TACLine_) {
    ASMLines asmlines;
    std::string str_label = TACLine_.src1.value;
    asmlines.push_back(construct_asm("jmp", str_label));
    return asmlines;
}

File: go_compiler/tcg/sentence_translator/GotoTranslator.py
from ...common.tac import TACLine
from ...tcg.SymbolManager import SymbolManager
from ...tcg.asm import ASMLines
from ...tcg.construct_asm import construct_asm
from ...tcg.sentence_translator.BaseTranslator import BaseTranslator


class GotoTranslator(BaseTranslator):
    def __init__(self) -> None:
        super().__init__()

    def SentenceTranslate_(
        self, SymbolManager_: SymbolManager, TACLine_: TACLine
    ) -> ASMLines:
        asmlines: ASMLines = ASMLines()
        str_label = TACLine_.src1.value
        asmlines.append(construct_asm("jmp", str_label))
        return asmlines
